tests/functional/i/inconsistent/inconsistent_returns_noreturn.py

```diff
@@ -53,3 +53,46 @@ def bug_pylint_4122_bis(s):
         return n
     except ValueError:
         parser_error_name('parser error')
+
+class ClassUnderTest:
+    def _no_return_method(self) -> typing.NoReturn:
+        sys.exit(1)
+
+    def _falsely_no_return_method(self) -> typing.NoReturn:
+        return 1
+
+    def _does_return_method(self) -> int:
+        return 1
+
+    def bug_pylint_8747(self, s: str) -> int:
+        """Every return is consistent because self._no_return_method hints NoReturn"""
+        try:
+            n = int(s)
+            if n < 1:
+                raise ValueError
+            return n
+        except ValueError:
+            self._no_return_method()
+
+    def bug_pylint_8747_wrong(self, s: str) -> int:  # [inconsistent-return-statements]
+        """Every return is not consistent because self._does_return_method() returns a value"""
+        try:
+            n = int(s)
+            if n < 1:
+                raise ValueError
+            return n
+        except ValueError:
+            self._does_return_method()
+
+    def bug_pylint_8747_underhanded(self, s: str) -> int:
+        """Every return is not consistent because self._falsely_no_return_method() returns a value.
+
+        Known false negative because we take the typing into account and not inference
+        contrary to what we usually do."""
```

Comment:
```suggestion
    def bug_pylint_8747_inference_trumps_incorrect_annotation(self, s: str) -> int:
```
I think the behavior as is is correct: why would we want to emit a message here? The slight asymmetry between whether to trust typing is fine with me: we use the typing only for preventing false positives, not when building the confidence to emit a message.

Reply:
I have no strong feelings about this. From my perspective, the behavior captured by that test falls outside the original scope of this PR.

Given that:

* Should I keep that test in the PR?
  * If so, @jacobtylerwalls would you prefer I commit your suggested rename?
* Otherwise, should I remove it? If this is something that requires further discussion/action, an issue could be made to track it.

Reply:
Let's keep the test. My suggestion is just to remove all the talk about the false negative and let the test stand for itself as expected behavior 👍

Reply:
Resolved by https://github.com/pylint-dev/pylint/pull/8750/commits/3c9a77c45689563f51f631ced8f3cfde0ad8c88b